src/bootstrap/step.rs

```diff
@@ -755,6 +766,7 @@ pub fn build_rules<'a>(build: &'a Build) -> Rules {
          .host(true)
          .only_host_build(true)
          .dep(|s| s.name("tool-cargo"))
+         .dep(|s| s.name("tool-rust-installer").stage(0))
```

Comment:
Oh here (and in other steps) I think you'll want to lock the `Step` here to `build.config.build` for both the host and the target triple, that's the architecture we want to build the tool for (as we're running it here, not packaging it)